src/Lexer/Lexer.php

```diff
@@ -160,8 +160,8 @@ private function generateRegexp(): string
 			self::TOKEN_PHPDOC_TAG => '@(?:[a-z][a-z0-9-\\\\]+:)?[a-z][a-z0-9-\\\\]*+',
 			self::TOKEN_PHPDOC_EOL => '\\r?+\\n[\\x09\\x20]*+(?:\\*(?!/)\\x20?+)?',
 
-			self::TOKEN_FLOAT => '(?:-?[0-9]++(_[0-9]++)*\\.[0-9]*(_[0-9]++)*+(?:e-?[0-9]++(_[0-9]++)*)?)|(?:-?[0-9]*+(_[0-9]++)*\\.[0-9]++(_[0-9]++)*(?:e-?[0-9]++(_[0-9]++)*)?)|(?:-?[0-9]++(_[0-9]++)*e-?[0-9]++(_[0-9]++)*)',
-			self::TOKEN_INTEGER => '-?(?:(?:0b[0-1]++(_[0-1]++)*)|(?:0o[0-7]++(_[0-7]++)*)|(?:0x[0-9a-f]++(_[0-9a-f]++)*)|(?:[0-9]++(_[0-9]++)*))',
+			self::TOKEN_FLOAT => '(?:[+-]?[0-9]++(_[0-9]++)*+\\.[0-9]*+(_[0-9]++)*+(?:e[+-]?[0-9]++(_[0-9]++)*+)?)|(?:[+-]?[0-9]*+(_[0-9]++)*+\\.[0-9]++(_[0-9]++)*+(?:e[+-]?[0-9]++(_[0-9]++)*+)?)|(?:[+-]?[0-9]++(_[0-9]++)*+e[+-]?[0-9]++(_[0-9]++)*+)',
```

Comment:
Some of these changes seem random to me and I don't know what to think about them. For example I don't know why there needs to be `+` after `*` - the last change on this line.

Your description says you want to add support for `+` after `e`. If so then please just add and test that.

Reply:
> `+` after `*`

I added it to help PCRE performance (in theory) as `++`/`*+` was already present. Is that ok?

> Your description says you want to add support for `+` after `e`. If so then please just add and test that.

This PR adds `+` sign support before any number (float/int) and after `E` as well. Is any test missing?

Reply:
Please just add support for `e+` and delete all the other changes.